### PR TITLE
[Skipper] Handle provide direct relative path in Skipper

### LINKED ISSUE #7467: RenameParamToMatchTypeRector not skipped
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Rector version | v0.14.2             |

From version 0.14.2 RenameParamToMatchTypeRector::class, $rectorConfig->skip is not working when processing single file.

config:
    $rectorConfig->skip([
        RenameParamToMatchTypeRector::class => [\_\_DIR\_\_ . '/Tests'],
    ]);

```
class Test
{
    public function test(UnitTester $I): void
    {
    }
}
```
vendor/bin/rector process --clear-cache --dry-run Tests/Test.php:

```
 class TransitionHelperCest
 {
-    public function test(UnitTester $I): void
+    public function test(UnitTester $unitTester): void
     {
     }
 }
```

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/7467

Step to reproduce:

```bash
cd e2e/skip-direct-file-path 
composer install
php ../e2eTestRunnerPath.php src/Test.php 
```

### CODE REVIEW

**Comment by samsonasik** [comment]
Fixed 🎉 /cc @AlexOstrovsky

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by samsonasik** [comment]
I added another failing test case for skip with path list only https://github.com/rectorphp/rector-src/pull/2921/commits/187d68582156d24510be4ad4cc9768e1960bdb91 , not by rector rule -> path.

with config:

```php
<?php

declare(strict_types=1);

use Rector\Config\RectorConfig;
use Rector\Naming\Rector\ClassMethod\RenameParamToMatchTypeRector;
use Rector\Set\ValueObject\SetList;

return static function (RectorConfig $rectorConfig): void {
    $rectorConfig->paths([__DIR__ . '/src']);

    $rectorConfig->skip([
        __DIR__ . '/src',
    ]);

    $rectorConfig->sets([SetList::NAMING]);
};

```

then use command:

```bash
php ../e2eTestRunnerPath.php -c rector_skip_path_list.php src/Test.php
```

**Comment by samsonasik** [comment]
Fixed 🎉

**Comment by samsonasik** [comment]
To run test locally:

```bash
cd e2e/skip-direct-file-path
composer install --ansi

# run e2e test direct path
php ../e2eTestRunnerPath.php src/Test.php

# run e2e test only directory
php ../e2eTestRunnerPath.php src

# run e2e test no path, look by rector.php
php ../e2eTestRunnerPath.php

# run e2e test with config use path lists
php ../e2eTestRunnerPath.php -c rector_skip_path_list.php src/Test.php
```

**Comment by samsonasik** [comment]
All checks have passed 🎉 🎉 🎉  @TomasVotruba I think it is ready.

File: packages/Skipper/Fnmatcher.php [hunk 1]
```diff
@@ -10,12 +10,41 @@ public function match(string $matchingPath, string $filePath): bool
...
+        if (is_file($realPathMatchingPath)) {
+            return $realPathMatchingPath === $realpathNormalizedFilePath;
+        }
+
+        // ensure add / suffix to ensure no same prefix directory
+        if (is_dir($realPathMatchingPath)) {
+            $realPathMatchingPath = rtrim($realPathMatchingPath, '/') . '/';
+        }
+
+        return str_starts_with($realpathNormalizedFilePath, $realPathMatchingPath);
```
Comment (TomasVotruba) [inline packages/Skipper/Fnmatcher.php]: The fnmatcher should handle only the * paths.
Rest shoule be managed by another matching case.

**Comment by TomasVotruba** [comment]
The e2e is too heavy test here. This should be unit tested only.

File: packages/Skipper/Fnmatcher.php (hunk 1 above)
Comment (samsonasik) [reply packages/Skipper/Fnmatcher.php]: Ok, I created `RealpatchMatcher` service for it https://github.com/rectorphp/rector-src/pull/2921/commits/30b4fcb76a43b4b55a9f69d61437b67250041994

**Comment by samsonasik** [comment]
@TomasVotruba the test is for relative path file location to check itself provided by command:

```bash
vendor/bin/rector process src/Test.php
```

that only happen when direct relative path provided in command itself.

which config can be:

```php
    $rectorConfig->skip([
        RenameParamToMatchTypeRector::class => [__DIR__ . '/src'],
    ]);
```

Or

```php
    $rectorConfig->skip([
         __DIR__ . '/src',
    ]);
```

so I think the way to test it is by using e2e test, Do we have example for unit test with path in command test?

File: packages/Skipper/RealpathMatcher.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        }
+
+        return str_starts_with($realpathNormalizedFilePath, $realPathMatchingPath);
+    }
+
+    private function normalizePath(string $path): string
+    {
+        return \str_replace('\\', '/', $path);
+    }
+}
```
Comment (TomasVotruba) [inline packages/Skipper/RealpathMatcher.php:42]: :clap:

**Comment by TomasVotruba** [comment]
> Do we have example for unit test with path in command test?

This is not neccesary. The path is only argument, that will be passed in configuration.

What is the input value that goes to a skipper component in case of running Rector?

```bash
vendor/bin/rector process src/Test.php?
```

That's what should be tested only in Skipper unit test.

**Comment by samsonasik** [comment]
@TomasVotruba Ok, moved to unit test with relative path provided in test provider https://github.com/rectorphp/rector-src/pull/2921/commits/eca1683af51980f71ed9ef4964157d8461fbdbf9 👍

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready now.

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Pull request merged**